### PR TITLE
feat(HNG-53): created the waitlist hero card component

### LINKED ISSUE #53: [FEAT] Waitlist > Create the waitlist hero card
### Description
Develop a reusable Waitlist Card component. This component will be used to display the services on the waitlist hero section of the website. The Card should be visually appealing, responsive, and should include the following elements:

### Acceptance Criteria
The Waitlist Card component is implemented according to the design
The component should be fully responsive and look good on all screen sizes.
Ensure accessibility standards are met and proper semantic HTML.

### Requirements
The component should receive the following props

- Icon
- Title
- Description
- Testing

Ensure the component renders correctly with all required elements.
Check that the component handles different types of content (long titles, missing icons).

### Tasks
- [ ] Develop the waitlist hero cards component based on the provided requirements.
- [ ] Implement the component in the codebase.
- [ ] Write unit tests to ensure the component works as expected.
- [ ] Test the component for responsiveness and accessibility.
- [ ] Update documentation with usage instructions.

### Figma link
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=518-7653&m=dev

### Screenshot
waitlist-hero-card

### PR DESCRIPTION
**Pull request by judeb21**
# Description
Github Issue 53: Closes #53  

#### Using the waitlist hero card component
It accepts 3 mandatory arguments and 1 optiona className as it's props, cardTitle, cardDescription and cardIcon.
- cardTitle: string
- cardDescription:  string
- cardIcon: string - Pass the absolute path of the icon to icon property

##### The commponent can be rendered as follows:
```
import WaitlistCard, { WaitlistCardProps } from '~/components/common/Waitlist/WaitlistCard';
const cardProp: WaitlistCardProps = {
  cardTitle: "Easy Customization",
  cardDescription: "Hello there!",
  cardIcon:"../../../public/Wailtist/vector-icon.svg"
}
<WaitlistCard {...cardProp} /> Use this way by spreading the card property
```

- OR

```
import WaitlistCard from '~/components/common/Waitlist/WaitlistCard';
<WaitlistCard
  cardTitle={"Easy Customization"}
  cardDescription={cardDescription}
  cardIcon={require("../../public/Wailtist/vector-icon.svg")}
/>
```

- Please do not alterate the Waitlist Hero Card Component, kindly reach out to me.

**Closes #53 **

# Changes proposed

## What were you told to do?
This PR is to create the wishlist hero card component

## What did you do?

- I created the waitlist hero card component
- I created a wishlist folder that will contain svg or images needed to complete the waitlist full page in the public directory
- I created a waitlist folder inside the common folder located in the component folder in the app directory
- I used [Shadcn card components](https://ui.shadcn.com/docs/components/card) in creating the card component and declared extra data types to accommodate the full card functionalities
- Created a waitlist card component test file inside a waitlist folder in the test folder of the app directory, where I wrote test cases to assert the functionality of the card component
- Added documentation in the contribution section of the readme file

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot 2024-07-19 at 01 51 11

### Eslint and Test Images
Screenshot 2024-07-20 at 18 57 13

Screenshot 2024-07-20 at 18 58 32

Screenshot 2024-07-20 at 18 58 53

### CODE REVIEW

File: README.md
Comment (Samadeen) [inline README.md:1]: Why is this here? You could have added it to your description

File: pnpm-lock.yaml
Comment (Samadeen) [inline pnpm-lock.yaml:1]: Why did you touch this??

**Review by Samadeen — CHANGES_REQUESTED** [review]

File: README.md
Comment (judeb21) [reply README.md:1]: @Samadeen I've added it to my description and removed it from the readme file

File: pnpm-lock.yaml
Comment (judeb21) [reply pnpm-lock.yaml:1]: I've rebased with the dev repo and fixed it. It's not longer part of the push @Samadeen. Thanks

**Comment by JulianaSau** [comment]
why are your force pushing changes?

File: src/components/common/Waitlist/WaitlistCard.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+type CardProperties = React.ComponentProps<typeof Card>;
+
+export type WaitlistCardProperties = CardProperties & {
+  cardTitle: string;
+  cardDescription: string;
+  cardIcon: string;
+};
+
+const waitlistBorderStyle = "border-[1px] border-[#525252] border-solid";
```
Comment (JulianaSau) [inline src/components/common/Waitlist/WaitlistCard.tsx]: use colors present in the tailwind config. dont use custom colors

File: src/components/common/Waitlist/WaitlistCard.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const ellipse =
+  "inline bg-[#F97316] h-[53px] w-[53px] rounded-[9999px] flex justify-center items-center mb-[12px]";
```
Comment (JulianaSau) [inline src/components/common/Waitlist/WaitlistCard.tsx]: view comment above on custom colors

File: src/components/common/Waitlist/WaitlistCard.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  className,
+  cardTitle,
+  cardDescription,
+  cardIcon,
+  ...properties
+}: WaitlistCardProperties) => {
+  return (
+    <Card
+      className={cn(
+        `w-[100%] bg-[#fafafa] md:w-[320px] ${waitlistBorderStyle}`,
```
Comment (JulianaSau) [inline src/components/common/Waitlist/WaitlistCard.tsx]: same here

File: src/components/common/Waitlist/WaitlistCard.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            height={22.5}
+            role="image"
+            src={cardIcon}
+            alt="Waitlist card icon"
+            title="Icon"
+          />
+        </div>
+        <CardTitle
+          role="cardTitle"
+          className={cn("text-[18px] text-[#525252]")}
```
Comment (JulianaSau) [inline src/components/common/Waitlist/WaitlistCard.tsx]: and here

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: src/components/common/Waitlist/WaitlistCard.tsx (hunk 1 above)
Comment (judeb21) [reply src/components/common/Waitlist/WaitlistCard.tsx]: Done @JulianaSau

File: src/components/common/Waitlist/WaitlistCard.tsx (hunk 2 above)
Comment (judeb21) [reply src/components/common/Waitlist/WaitlistCard.tsx]: Done @JulianaSau

File: src/components/common/Waitlist/WaitlistCard.tsx (hunk 3 above)
Comment (judeb21) [reply src/components/common/Waitlist/WaitlistCard.tsx]: Done @JulianaSau

File: src/components/common/Waitlist/WaitlistCard.tsx (hunk 4 above)
Comment (judeb21) [reply src/components/common/Waitlist/WaitlistCard.tsx]: All done @JulianaSau

**Review by mrcoded — APPROVED** [review]
